snapshot/policy/retention_policy.go

```diff
@@ -160,6 +168,14 @@ func (r *RetentionPolicy) getRetentionReasons(i int, s *snapshot.Manifest, cutof
 			continue
 		}
 
+		if *r.KeepMinDays != *newOptionalInt(0) {
+			if !s.StartTime.Before(cases[6].cutoffTime) {
+				keepReasons = []string{"Inside min retention days"}
+			}
+
+			continue
+		}
```

Comment:
I used `c.max = r.KeepMinDays` as the variable `max` already exists, Or else I think we need to add a new variable to the `case `struct. Can you suggest any changes

Reply:
I believe that cutoff time here will serve your purpose (do double check that though), it will retain all snapshots within the calculated period, and if `c.max` is `MAX_INT`, it'll do so for any number of such snapshots.

Reply:
I have moved this logic out of `for` loop